lib/report_formatter/timeline.rb

```diff
@@ -52,6 +52,80 @@ def to_cid(id)
       ApplicationRecord.compress_id(id)
     end
 
+    def tl_href(col, rec, text, ems_cloud, ems_container)
+      case col
```

Comment:
A hash would need to compute all values (some possibly hitting the DB, eg. `rec.ext_management_system.id`), and then use only one, no?

How about `send("tl_href_#col")` to have each case in a separate method? (will need to guard by responds_to? or rescue logic.)
Though it's unwise to listen to me suggesting meta-programming, given my near-zero ruby experience ;-)